### ISSUE TITLE
making repositioning of context tag and project tag optional

### ISSUE THREAD

**Issue by Tobida** [ransome1/sleek#68]
I love to use todo.txt in a natural way of language:

> discuss with @George about +superproject concerning the aspect of unicorns

This is in consensus with the todo.txt rules i read: "Contexts and Projects may appear anywhere in the line after priority/prepended date"

In my sleek this will show up like this, the sentence is destroyed 

> discuss with about concerning the aspect of unicorns @George +superproject

I fully respect that in your perspective this is not a bug but a feature. 

I would like to use it without the feature of putting tags at the and of the line - perhaps there could be an option  to switch this behaviour off?

Thanks!

**Comment by ransome1** [comment]
Hi @Tobida . I'm totally with you on this topic and would prefer it the way you describe it too. But this is actually not up to me, as I use a JavaScript library to do the parsing of the text into proper todo.txt objects. And while it does a great job in general, it doesn't persist the position of projects and contexts after generating the object, but sorts these attributes in it's own way. 

But in fact there is a feature request for exactly that feature at the project: https://github.com/jmhobbs/jsTodoTxt/issues/5

You can give it a push over there and as soon as they implement it, I will include it into sleek.

**Comment by 9z0** [comment]
A similar repositioning request: is it possible to have a function to sort the projects/contexts *inside* the todo alphabetically?
For example,
```
(A) todo +b +a @context_b @context_a
```
running the function makes it
```
(A) todo +a +b @context_a @context_b
         ^     ^
```

**Comment by dnngll** [comment]
+1, super important imho

**Comment by aubreyz** [comment]
Agree here - I am never likely to use  to use Slick seriously if it messes with the Todo file and changing the order of the words is a serious problem. One if the points if the protocol is to be  be able to use natural text.

**Comment by ransome1** [comment]
I'm closing here, the issue will further discussed in https://github.com/ransome1/sleek/issues/68

**Comment by dnngll** [comment]
> I'm closing here, the issue will further discussed in #68

Isn't this #68? 🧐

**Comment by dnngll** [comment]
Hello @ransome1 ,

since https://github.com/jmhobbs/jsTodoTxt/issues/5 has been getting some attention lately, I wondered if the matter could be addressed in Sleek again. You closed https://github.com/ransome1/sleek/issues/68 in September, but I didn't find an open issue with that problem (https://github.com/ransome1/sleek/issues/92, https://github.com/ransome1/sleek/issues/183, https://github.com/ransome1/sleek/issues/216, https://github.com/ransome1/sleek/issues/220, https://github.com/ransome1/sleek/issues/306, https://github.com/ransome1/sleek/issues/370 and https://github.com/ransome1/sleek/issues/411 are all closed).

Things have been getting better, since ticking of a box now only changes that very line, and not the whole file any more. But the resorting on the ticked todo is still happening unfortunately.

Sleek is getting better and better every week, I am really hoping to be able to use it to tick off todos as well soon : )

Please let me know what you think.

Thanks!

**Comment by ransome1** [comment]
Yes, as soon as jsTodoTxt is finishing the rewrite, sleek will adapt to it and hopefully this will solve these issues. 

But for now the latest version of jsTodoTxt is alpha and for almost two weeks there has been no progress. I will start implementing as soon as a stable version has been released and the documentation is complete :)

**Comment by ransome1** [comment]
> > I'm closing here, the issue will further discussed in #68
> 
> 
> 
> Isn't this #68? 🧐

Indeed it is. There was no good reason to close here. I'll reopen.

**Comment by dnngll** [comment]
Wow,

that's a swift response, thanks @ransome1 ! Also thanks for reopening. You have no idea how much I am looking forward to using Sleek as my daily driver.

**Comment by intrepidsilence** [comment]
I think this issue is fixed in the upstream Library. Any word on when this might get integrated?

**Comment by ransome1** [comment]
> I think this issue is fixed in the upstream Library. Any word on when this might get integrated?

The author started a complete rewrite and as of mid of October and so far I have only seen a very early alpha.

**Comment by ransome1** [comment]
I'm currently rewriting sleek and during the process I switched to the current Alpha branch of jsTodoTxt. Thanks to the good work upstream, the future implementation will render `(C) Lorem +ipsum dolor @amet lorem due:2023-11-11 amet dolor pm:2 lorem ipsum dolor amet` like this:

![Bildschirmfoto 2023-07-16 um 12 09 22](https://github.com/ransome1/sleek/assets/11188741/29e6a3fd-a0f2-4d3d-90bc-19d50324942d)

I will soon publish a first developer preview for testing purposes.

**Comment by intrepidsilence** [comment]
Excellent!  Whoohoo! Thank you!

**Comment by dnngll** [comment]
> I'm currently rewriting sleek and during the process I switched to the current Alpha branch of jsTodoTxt. Thanks to the good work upstream, the future implementation will render `(C) Lorem +ipsum dolor @amet lorem due:2023-11-11 amet dolor pm:2 lorem ipsum dolor amet` like this:
> 
> ![Bildschirmfoto 2023-07-16 um 12 09 22](https://user-images.githubusercontent.com/11188741/253797891-29e6a3fd-a0f2-4d3d-90bc-19d50324942d.png)
> 
> I will soon publish a first developer preview for testing purposes.

Oh wow, can't wait! Thank you so much. So looking forward to using Sleek as my one-stop-solution!

**Comment by ransome1** [comment]
@intrepidsilence @Tobida @dnngll @aubreyz @9z0 this should have solved in the very early developer preview of 2.0. Feel free to test it: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev1

**Comment by dnngll** [comment]
Thank you so much! I just tried `sleek-2.0.0-dev1-win.zip`, but I am unable to open an existing file:

![2023 07 18, 18 54 34](https://github.com/ransome1/sleek/assets/12801866/5d902cf5-8a16-4775-8173-8629bd5d109b)

![2023 07 18, 18 54 43](https://github.com/ransome1/sleek/assets/12801866/edae1ee4-f0ca-4ac6-ba5d-affda4e8467b)

![2023 07 18, 18 54 59](https://github.com/ransome1/sleek/assets/12801866/91add6c2-f817-43d2-b275-134693f9dc5b)

Only after restarting Sleek, I see this (todo.txt is full of tasks):

![2023 07 18, 18 55 58](https://github.com/ransome1/sleek/assets/12801866/683328db-154f-43fe-ad7e-75e433ab65f9)

I didn't change any settings, just unzipped and started it. Am I doing sth. wrong here?

Thanks!

**Comment by intrepidsilence** [comment]
Awesome! Thank you so much! Works for me (macOS ARM).

**Comment by ransome1** [comment]
![image001](https://github.com/ransome1/sleek/assets/11188741/947eef18-2297-48f3-a5d4-f6477acc8d2a)

I tried the release you picked on my Windows machine and it worked. However I don't think you're doing anything wrong here, it must just work on all systems equally. This version also doesn't touch your current configuration file, so that shouldn't be the problem neither.

I could recreate the view you see, after I renamed/deleted the file I already added. Are you running into the issue with the .exe or Setup release files, as well?

**Comment by ransome1** [comment]
> Thank you so much! I just tried `sleek-2.0.0-dev1-win.zip`, but I am unable to open an existing file:
> 
> ![2023 07 18, 18 54 34](https://user-images.githubusercontent.com/12801866/254338177-5d902cf5-8a16-4775-8173-8629bd5d109b.png)
> 
> ![2023 07 18, 18 54 43](https://user-images.githubusercontent.com/12801866/254338224-edae1ee4-f0ca-4ac6-ba5d-affda4e8467b.png)
> 
> ![2023 07 18, 18 54 59](https://user-images.githubusercontent.com/12801866/254338298-91add6c2-f817-43d2-b275-134693f9dc5b.png)
> 
> Only after restarting Sleek, I see this (todo.txt is full of tasks):
> 
> ![2023 07 18, 18 55 58](https://user-images.githubusercontent.com/12801866/254338501-683328db-154f-43fe-ad7e-75e433ab65f9.png)
> 
> I didn't change any settings, just unzipped and started it. Am I doing sth. wrong here?
> 
> Thanks!

When you click on "File" in that top bar menu, do you see your todo file listed there?

**Comment by dnngll** [comment]
> When you click on "File" in that top bar menu, do you see your todo file listed there?

I do:

![image](https://github.com/ransome1/sleek/assets/12801866/d48dd58a-983e-4808-b3af-b23a3a46810c)

**Comment by dnngll** [comment]
I just tried the same as above, but with "sleek-2.0.0-dev3-win.zip". I deleted everything in the folder "Sleek" and copied the contents of the new .zip in there. Right at first start I see this:

![2023 07 25, 19 33 41](https://github.com/ransome1/sleek/assets/12801866/7d2cf10f-ec02-440e-9692-21aa6b4c14c4)

I did not tell Sleek (dev3) where to look for the file. I only set this up before in (the now deleted) Sleek (dev1) 🧐

Besides this, the todo is not loading (same as before on dev1)...

**Comment by ransome1** [comment]
@dnngll thanks for looking into again. I was able to recreate it on Windows. I will propably have it fixed with the next developer preview.

**Comment by dnngll** [comment]
> @dnngll thanks for looking into again. I was able to recreate it on Windows. I will propably have it fixed with the next developer preview.

@ransome1

you're very welcome - thank **you**!

And: Were you talking about fixing the "file"-problem, or the "portable app saves settings somewhere else"-problem?

**Comment by ransome1** [comment]
> > @dnngll thanks for looking into again. I was able to recreate it on Windows. I will propably have it fixed with the next developer preview.
> 
> @ransome1
> 
> you're very welcome - thank **you**!
> 
> And: Were you talking about fixing the "file"-problem, or the "portable app saves settings somewhere else"-problem?

Why exactly it crashed in the Dev1 I still havn't figured out. But as for the reason why it still doesn't work for Dev 3 is most likely the old config files from Dev 1. In these something essential is missing, which made sleek not start on my Windows machine.

Can you please delete the `userData` folder, which is located here: `C:\Users\YOURUSERNAME\AppData\Roaming\sleek\userData`

After doing this, a fresh default configuration will be created, which hopefully solves the issue on your side.

**Comment by dnngll** [comment]
I'll do you one better: Just restored Windows from an image where Sleek has never been started on, plus deleted all files in the portable (dev3) directory and unzipped again.

This is what I get without restarting Sleek:

![2023 07 26, 18 26 27](https://github.com/ransome1/sleek/assets/12801866/f027192b-5a32-44b7-a9a3-6e25ea8f2666)

![2023 07 26, 18 26 42](https://github.com/ransome1/sleek/assets/12801866/3f96e03a-5374-4095-98fa-e585c18ec2c3)

![2023 07 26, 18 27 00](https://github.com/ransome1/sleek/assets/12801866/47c4cab3-bc1c-40ae-85d1-21f9a4c05687)

![2023 07 26, 18 27 16](https://github.com/ransome1/sleek/assets/12801866/87847326-cec3-4ef2-ba84-d8c0526de5ba)


And this after restarting sleek:

![2023 07 26, 18 27 40](https://github.com/ransome1/sleek/assets/12801866/846f0ca6-9b05-4f01-8cca-5d6d0585f97b)

So I guess it'll be best to wait for dev4, huh?

**Comment by ransome1** [comment]
Dev 4 won't solve it, unless we figure out what leads to this behavior on your side. 

Do you know your way around PowerShell? I think it comes installed on a Windows machine. If you start sleek from shell, it will most likely throw you an error, with which we could work. You would need to navigate to the folder, in which sleek.exe is located. Then just type `.\sleek.exe`. This should start the app and in PowerShell you will see some logs. Hopefully there will be an error popping up.

**Comment by dnngll** [comment]
Sure. This is starting Sleek dev3 (after the userfolder has been deleted), and then opening the "! ToDo.txt":

```
PS Z:\sleek> .\sleek.exe
PS Z:\sleek>
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: Z:\Sleek\! ToDo.txt
FileWatcher.ts: Initial scan complete. Ready for changes
TypeError: Cannot read properties of null (reading '2')
    at t.Item.parse (Z:\sleek\resources\app.asar\dist\main\main.js:14:248482)
    at new t.Item (Z:\sleek\resources\app.asar\dist\main\main.js:14:248288)
    at Z:\sleek\resources\app.asar\dist\main\main.js:14:326887
    at Array.map (<anonymous>)
    at Z:\sleek\resources\app.asar\dist\main\main.js:14:326866
    at _ (Z:\sleek\resources\app.asar\dist\main\main.js:14:327359)
```

I hope, that helps?

**Comment by dnngll** [comment]
Not until now.

This is what I get when starting Sleek dev3 (after the userfolder has been deleted), and then creating a new file from within Sleek, and then adding 2 todo's:

```
PS Z:\sleek> .\sleek.exe
PS Z:\sleek>
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: Z:\Sleek\todo.txt
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
File Z:\Sleek\todo.txt has been changed
ipcEvents.ts: FIX ME!
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
File Z:\Sleek\todo.txt has been changed
ipcEvents.ts: FIX ME!
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for files
```

Or as a Screenshot:

![image](https://github.com/ransome1/sleek/assets/12801866/4ad30dc5-378c-4264-abad-0947bac5a0ee)

Looks better definitely : )

**Comment by dnngll** [comment]
No, I wouldn't say so. I got `€äöüÄÖÜß` in it, but the file worked on the previous version of sleek (and a couple of other tools I am using.

I will delete half of it and check, then try the other half and so on... will get back to you in a bit!

**Comment by ransome1** [comment]
Thanks for figuring this out together

**Comment by dnngll** [comment]
Nonono thank **you** for your work!!! 😊

I think we're getting somewhere. `! ToDo 1.txt` does not work, `! ToDo 2.txt` does. Only difference is the last empty line:

![image](https://github.com/ransome1/sleek/assets/12801866/09e37e45-c6b7-4267-abe1-e387d83923e9)

---

However: Just deleting the last line from my original ToDo won't fix it until I only have 1 ToDo left... Doesn't matter which one. This file:

![image](https://github.com/ransome1/sleek/assets/12801866/bc7cc09b-c416-430a-a1f7-5017bee62f9a)

does not work!

**Comment by ransome1** [comment]
I tried to reproduce it the way you describe it. Created the same file, I have a Windows machine running, I use Sublime Text and the sleek Dev3. But I simply cannot reproduce it :/

If you say "it does not work", you mean the screen stays white and the same error pops up in PowerShell? 

Also, would you mind sharing the `config.json` file in `userData` folder? It doesn't contain any sensitive data.

**Comment by dnngll** [comment]
Would you like to see my shared screen? Maybe that's a little easier to see
what I'm doing, and if I'm making any mistakes... I'm using Senfcall, it's
browser based, no need to install anything. We could do it right now.

**Comment by dnngll** [comment]
Would you like to see my shared screen? Maybe that's a little easier to see what I'm doing, and if I'm making any mistakes... I'm using Senfcall, it's browser based, no need to install anything. We could do it right now.

**Comment by dnngll** [comment]
Would you like to see my shared screen? Maybe this way it's a little easier to see what I'm doing, and if I'm making any mistakes... I'm using Senfcall, it's browser based, no need to install anything. We could do it right now.

**Comment by dnngll** [comment]
Would you like to see my shared screen? Maybe this way it's a little easier to see what I'm doing, and whether I'm making any mistakes... I'm using Senfcall, it's browser based, no need to install anything. We could do it right now.

**Comment by dnngll** [comment]
Come to

https://public.senfcall.de/sleek

if you're interested in a quick live demo...

**Comment by dnngll** [comment]
> if you're interested in a quick live demo...

Hmm you seem to not have gotten my last message yet. Will leave the room again... We can still do this tomorrow if you think it might help.

---

> If you say "it does not work", you mean the screen stays white and the same error pops up in PowerShell?

I didn't start it with powershell this time. But the result is the same as above (https://github.com/ransome1/sleek/issues/68#issuecomment-1652146774).

---

> Also, would you mind sharing the `config.json` file in `userData` folder? It doesn't contain any sensitive data.

Sure. That's the config after deleting the userfolder, starting and closing Sleek:

```
{
  "sorting": [
    "priority",
    "due",
    "projects",
    "contexts",
    "t",
    "completed",
    "created"
  ]
}
```

And that's the config after reopening "the Sleek from above" (without deleting the userfolder this time), adding the todo-file and then closing it:

```
{
	"sorting": [
		"priority",
		"due",
		"projects",
		"contexts",
		"t",
		"completed",
		"created"
	],
	"files": [
		{
			"active": true,
			"path": "Z:\\Sleek\\! ToDo.txt",
			"filename": "! ToDo.txt"
		}
	]
}
```
I hope this helps?

I will go to bed now, have a good night!

**Comment by ransome1** [comment]
> Come to
> 
> https://public.senfcall.de/sleek
> 
> if you're interested in a quick live demo...

Oh no, sorry. I was already off the computer when you wrote this. But much appreciated!

I think now I have everything you could provide and it's up to me now to reproduce it. I'll keep you posted.

**Comment by dnngll** [comment]
> I think now I have everything you could provide

Perfect, let me know if you need anything more!

**Comment by dnngll** [comment]
Hey @ransome1 

just a quick info: Using `sleek-2.0.0-dev5-win.zip` `[dad1345a10ef6308a4dbcd65af405068]` (on a brand new system, where sleek has never been started on) I still have the same problem as above... The offer about showing you my screen is still there if that helps.

**Comment by ransome1** [comment]
@dnngll It really is a mystery to me.

Although I'm working on a Mac, I have a Windows system here and the Dev5 works without any issue. Same goes for Linux.

With one of the earlier developer previews I ran into a problem, similar to what you were experiencing. That was due to an incompatibility of the config files from different versions.

From Dev4 on this should not be an issue any longer, because for every new developer preview version the config is freshly created and put into its own versionized folder.

If you start sleek.exe from Power Shell, does it give you a new/different error or anything?

Also can I ask you to completely move or delete your config folder, just to make sure it is not the configs' fault?

If you start sleek with the Power Shell it will output the location where the config folder is located. It should be something like: `C:\Users\[YOURUSERNAME]\AppData\Roaming\sleek\userData2.0.0-dev5 `

**Comment by ransome1** [comment]
And one more question: Is there a way you could create a todo file, which crashes sleek on your side and send it to me? One with just bogus data. You can send it here: ransome@duck.com

**Comment by ransome1** [comment]
@dnngll if your issue is related to the filewatcher, which is being used in sleek, we should give this latest developer preview a chance: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev7

I tweaked the configuration a little bit. I'm still curious about a "broken" text file from you, in order to fully reproduce this.

**Comment by dnngll** [comment]
Hey @ransome1 

using the now new `sleek-2.0.0-dev7-win.zip` `[ace17a1a2cf7bd4daa08f12dc24205b9]`.

---

> From Dev4 on this should not be an issue any longer, because for every new developer preview version the config is freshly created and put into its own versionized folder.

Yeah I saw that. Folder `C:\Users\YOURUSERNAME\AppData\Roaming\sleek` is deleted by me before every new try below.

---

> If you start sleek.exe from Power Shell, does it give you a new/different error or anything?

This is sleek using file `1 working.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev7
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\1 working.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
```

This is sleek using file `2 not working.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev7
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\2 not working.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
(node:12152) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading '2')
    at t.Item.parse (B:\sleektest\resources\app.asar\dist\main\main.js:14:228532)
    at new t.Item (B:\sleektest\resources\app.asar\dist\main\main.js:14:228338)
    at B:\sleektest\resources\app.asar\dist\main\main.js:14:456602
    at Array.map (<anonymous>)
    at T (B:\sleektest\resources\app.asar\dist\main\main.js:14:456581)
    at R (B:\sleektest\resources\app.asar\dist\main\main.js:14:458683)
    at async IpcMainImpl.ee (B:\sleektest\resources\app.asar\dist\main\main.js:14:466576)
(Use `sleek --trace-warnings ...` to show where the warning was created)
(node:12152) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

---

> And one more question: Is there a way you could create a todo file, which crashes sleek on your side and send it to me? One with just bogus data. You can send it here: [ransome@duck.com](mailto:ransome@duck.com)

Both files `[41d7286db70985e7ac05a1a5afa67e3a]` are available at https://filetransfer.io/data-package/kLgvrjYA

---

The difference between `1 working.md` and `2 not working.md`: "1" only has one line (one ToDo)... 🙄

---

I hope that helps? Thanks again for your effort!

**Comment by dnngll** [comment]
> @dnngll if your issue is related to the filewatcher, which is being used in sleek, we should give this latest developer preview a chance: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev7
> 
> I tweaked the configuration a little bit. I'm still curious about a "broken" text file from you, in order to fully reproduce this.

Funny you **just** posted this... But I did use dev7 above, phew 😇

**Comment by ransome1** [comment]
Hehe :)

The good news is, I can reproduce the issue with the non working file in my working environment. Let's see if I can find the root cause here.

**Comment by dnngll** [comment]
> Hehe :)
> 
> The good news is, I can reproduce the issue with the non working file in my working environment. Let's see if I can find the root cause here.

PERFECT!!!

Looking at how you've handled issues in the past, I think it is safe to say that sleek will becomes my daily driver very soon 😊🥳

You have no idea how long I've been hoping for this to happen! Thank you so much!

**Comment by dnngll** [comment]
> 

PERFECT!!!

Looking at how you've handled issues in the past, I think it is safe to say that sleek will become my daily driver very soon 😊🥳

You have no idea how long I've been hoping for this to happen! Thank you so much!

**Comment by dnngll** [comment]
> Hehe :)
> 
> The good news is, I can reproduce the issue with the non working file in my working environment. Let's see if I can find the root cause here.

PERFECT!!!

Looking at how you've handled issues in the past, I think it is safe to say that sleek will become my daily driver very soon 😊🥳

You have no idea how long I've been hoping for this to happen! Thank you so much!

**Comment by dnngll** [comment]
Thank you for looking into it! That's so weird, I can't be the only one having this problem...

> I cannot find anything in that line, which looks unusual.

That's what I mean... Nothing special to the file :(

> That's something I will pass on to the `jstodotxt` project.

Thanks. So that's nothing you can fix on our side, right?

> As for now I added an error handling, which, will catch errors, when `jstodotxt` cannot parse a line and will skip that line. This doesn't solve our problem, as it would simply cut out your broken todo line. But at least render the rest.

I'll try my "real" todo-file in dev8 and will let you know how many lines are skipped.

> I don't know if you already mentioned it, but can you tell me what other apps (Todo apps, editors, or anything which manipulates the files) you use to work on your todo files?

On Windows I am using todotxt.net (https://github.com/jdckr/todotxt.net), Sublime Text portable, a self-made batch to add todos, a self-made ahk-script to add todos, Obsidian and Windows notepad.exe.

On Android I am using Simpletask and Obsidian

**Comment by ransome1** [comment]
I think if I find out what exactly it is, that is making jsTodoTxt throw the error, I can intercept it. If it's some hidden character, I can cut it out. I will investigate it.

**Comment by dnngll** [comment]
Okay, thanks!

**Comment by ransome1** [comment]
I havn't found a solution yet, but I think I can say with some certainty, that the problem lies in the last character of the 1st line in `2 not working.md`
<img width="249" alt="Bildschirmfoto 2023-09-01 um 20 38 39" src="https://github.com/ransome1/sleek/assets/11188741/c154ded0-d27d-4303-b402-aefc8291a0bb">

It is something read by JavaScript as a whitespace character. If I simply cut the last character out of the string, this does not cut out the `z`, which should be the last character. And if I then send it to the jstodotxt module, the todo is being parsed.

I tried to find out what this character is, but were not successful yet. It is not a new line `\n` neither is it an empty space.

Could you check, if your custom script does add some unique character at the end of the line? Maybe it tries to enforce a new line or something like that.

**Comment by ransome1** [comment]
@dnngll although it won't really help you, you can try out the latest developer preview: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev8

If a line in the file contains something, which `jstodotxt` cannot 'digest', sleek will skip it. 

But I assume, that all the lines in your file will contain this mysterious character and it might cut out all your todos :/

**Comment by dnngll** [comment]
Hey @ransome1 ,

thank you for your reply. I just got back and tried something new, which will address your point in https://github.com/ransome1/sleek/issues/68#issuecomment-1703192637 .

---

I created a new file through the Windows context menu "New | Text Document" and named it `3 working notepad.md`. I used Windows Notepad and typed the content by hand. This is the content of file `3 working notepad.md`:

```
2121-09-13 +dede +frfr gtgt @hzhz
```

Or as a screenshot, cursor on end:

![2023 09 13, 17 06 42](https://github.com/ransome1/sleek/assets/12801866/b9c33e40-1e7d-44e8-a5cf-4d02c2ee98b1)

---

Then I duplicated that file, renamed it into `4 not working notepad.md`, opened it with Notepad again and added a new line at the end (just by pressing enter). This is the content of file `4 not working notepad.md`:

```
2121-09-13 +dede +frfr gtgt @hzhz

```

Or again as a screenshot, cursor on end:

![2023 09 13, 17 06 51](https://github.com/ransome1/sleek/assets/12801866/be3695a2-573d-4800-8608-9a6dc846c887)

---

These are the results of Powershell on `3 working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev7
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\3 working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
```

---

These are the results of Powershell on `4 not working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev7
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\4 not working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
(node:14588) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading '2')
    at t.Item.parse (B:\sleektest\resources\app.asar\dist\main\main.js:14:228532)
    at new t.Item (B:\sleektest\resources\app.asar\dist\main\main.js:14:228338)
    at B:\sleektest\resources\app.asar\dist\main\main.js:14:456602
    at Array.map (<anonymous>)
    at T (B:\sleektest\resources\app.asar\dist\main\main.js:14:456581)
    at R (B:\sleektest\resources\app.asar\dist\main\main.js:14:458683)
    at async IpcMainImpl.ee (B:\sleektest\resources\app.asar\dist\main\main.js:14:466576)
(Use `sleek --trace-warnings ...` to show where the warning was created)
(node:14588) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

---

That's so weird, isn't it? You could simply create these files yourself, however I uploaded then just in case you want to check them out: https://filetransfer.io/data-package/EjleeshY

---

I used dev7 (`sleek-2.0.0-dev7-win.zip` `[ace17a1a2cf7bd4daa08f12dc24205b9]`), because I didn't want the errors to be cut out.

Also, I always deleted the userfolder before.

---

I hope you can make any sense of this? I will try the same with the current dev now - just in case! Will get back to you asap...

**Comment by dnngll** [comment]
That took forever, since I'm on a very slow connection currently :(

---

Using dev9 (`sleek-2.0.0-dev9-win.zip` `[214ead1e2e776184a55da23d1812a5a9]`), always deleting userfolder, using the files from above (https://github.com/ransome1/sleek/issues/68#issuecomment-1717859605):

---

These are the results of Powershell on `3 working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev9
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set isNavigationHidden to false
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\3 working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
```

Plus, a screenshot:

![2023 09 13, 17 39 08](https://github.com/ransome1/sleek/assets/12801866/7c42b7e1-4ece-43ed-a408-c87d65d79837)

---

These are the results of Powershell on `4 not working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev9
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set isNavigationHidden to false
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\4 not working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
TypeError: Cannot read properties of null (reading '2')
    at t.Item.parse (B:\sleektest\resources\app.asar\dist\main\main.js:14:228532)
    at new t.Item (B:\sleektest\resources\app.asar\dist\main\main.js:14:228338)
    at B:\sleektest\resources\app.asar\dist\main\main.js:14:457396
    at Array.map (<anonymous>)
    at D (B:\sleektest\resources\app.asar\dist\main\main.js:14:457371)
    at j (B:\sleektest\resources\app.asar\dist\main\main.js:14:459144)
    at async IpcMainImpl.ne (B:\sleektest\resources\app.asar\dist\main\main.js:14:468003)
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationHidden
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
```

Plus, a screenshot again:

![2023 09 13, 17 40 04](https://github.com/ransome1/sleek/assets/12801866/bed978be-2320-4388-9b20-fed1eea407e0)

---

Does that help?

**Comment by dnngll** [comment]
Uh and just for shits and giggles I opened my real todo.txt in dev9 (`sleek-2.0.0-dev9-win.zip` `[214ead1e2e776184a55da23d1812a5a9]`).

Here's the file in Sublime (obfuscated):

![2023 09 13, 17 48 31](https://github.com/ransome1/sleek/assets/12801866/c693c2a8-1230-4445-8488-d388ff4268a2)

And here's the file in Sleek:

![2023 09 13, 17 48 10](https://github.com/ransome1/sleek/assets/12801866/69d516c0-ebf7-4fb4-ba5b-6c3911d44209)

Like you said, everything is filtered out... 😢

**Comment by dnngll** [comment]
TLDR: IT'S WORKING!!! 🎉

---

I see you just released dev10 - wow! Here are the results using dev10 (`sleek-2.0.0-dev10-win.zip` `[2c5a9b8e21d8472e5021e36cd0df1950])`, always deleting userfolder, using the files from above (https://github.com/ransome1/sleek/issues/68#issuecomment-1717859605):

---

These are the results of Powershell on `3 working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev10
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\3 working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
```

---

These are the results of Powershell on `4 not working notepad.md`:

```
PS B:\sleektest> .\sleek.exe
PS B:\sleektest>
config.ts: sleek userdata is located at: C:\Users\d21\AppData\Roaming\sleek\userData2.0.0-dev10
main.ts: Main window has been created successfully
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Set accordionOpenState to true,true,true,false,false,false,false,false,false
ipcEvents.ts: Set drawerWidth to 500
ipcEvents.ts: Set sorting to [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]
ipcEvents.ts: Set isDrawerOpen to false
ipcEvents.ts: Set isSearchOpen to false
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
File.ts: File watchers created
File.ts: File added, restarting file watchers
ipcEvents.ts: Received config value for files
FileWatcher.ts: New file added: B:\sleektest\4 not working notepad.md
FileWatcher.ts: Initial scan complete. Ready for changes
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
ipcEvents.ts: Received config value for appendCreationDate
ipcEvents.ts: Received config value for convertRelativeToAbsoluteDates
ipcEvents.ts: Received config value for notificationsAllowed
ipcEvents.ts: Received config value for drawerWidth
ipcEvents.ts: Received config value for accordionOpenState
ipcEvents.ts: Received config value for files
ipcEvents.ts: Received config value for isDrawerOpen
ipcEvents.ts: Received config value for isSearchOpen
ipcEvents.ts: Received config value for sorting
ipcEvents.ts: Received config value for isNavigationOpen
ipcEvents.ts: Received config value for colorTheme
ipcEvents.ts: Received config value for shouldUseDarkColors
ipcEvents.ts: Received config value for showFileTabs
```

---

I'm really curious - what was the issue?

Again, thank you so much. I am really looking forward to using Sleek now 😁

**Comment by dnngll** [comment]
I see! I find it really weird that I am the only one who was complaining about this. Seems no one else is on Windows... Or trying out the dev-versions...

I am just playing around with the (broken) dev10 (`sleek-2.0.0-dev10-win.zip` `[2c5a9b8e21d8472e5021e36cd0df1950]`) that I downloaded before, but will update to the newer version later as soon as you release it.

**Comment by dnngll** [comment]
You're doing something with the settings, right? Hardly anything there until dev7 (I didn't test dev8)

![2023 09 13, 18 46 30](https://github.com/ransome1/sleek/assets/12801866/c16fb1bb-0516-4af4-852a-b8aedb2292d3)
(dev7)

and not much more there in dev9 or the defective dev10...

![2023 09 13, 18 47 24](https://github.com/ransome1/sleek/assets/12801866/a50e42c2-2bf1-439d-a372-44aa04aaff95)
(dev9)

**Comment by dnngll** [comment]
Totally disregard my last message. After playing around with that version for a bit, I realized that hardly anything is working yet.

I should have considered your info about the nightlies: "_This is an early developer preview with some basic functionality implemented, although many features are still missing._"

Looking forward to all the new features making it into the stable 👐 and will of course keep on testing future dev-versions.

Thanks

**Comment by dnngll** [comment]
I hope I didn't offend you by saying this, sorry if I did, didn't mean to. Here's what did not work:

- After ticking off a ToDo, all other ToDos are not shown anymore (yet are still in the file). Maybe that has to do with the fact, that the ToDo was saved using CRLF before. After Sleek touches the file, it is saved using LF. When manually saving the file using CRLF again in another editor, all ToDos are displayed again in Sleek
- Only 4 options available in Settings
- Dark Theme will only apply to the title bar
- Applying Dark Theme is not saved, it will only work the first time. After restarting Sleek it will not show and cannot be set again. To set again one has to cheat: Sleek-Folder and User-Folder have to be deleted
- Can't filter by "has no" Project- or Context (or I can't find it)

So it was not really usable for me right now, which is why I thought that not everything has been implemented in these builds...

Hope I could help!

**Comment by ransome1** [comment]
> So it was not really usable for me right now, which is why I thought that not everything has been implemented in these builds...

It is not suppose to be usable for production use: https://github.com/ransome1/sleek/discussions/501

I updated the release description texts to further clear up, what is happening in this project right now.

Also let me close this thread for now, since we more or less hijacked it a little bit for something not related to the initial requirement. The line break issue should be resolved in the upcoming developer preview 10 and if it is not we should rather open a separate bug task.

**Comment by dnngll** [comment]
> It is not suppose to be usable for production use: #501
>
> I updated the release description texts to further clear up, what is happening in this project right now.

Yeah, my bad. I was so excited that you fixed https://github.com/ransome1/sleek/issues/68 and wanted to use Sleek as my daily driver right away. So in my fantasy it was more of a "test new features in an upcoming version", than a complete rewrite. I know better now 😊 

Really looking forward to "_Once this foundational work is complete, there is room for new features. Hopefully developed not just by me. I sincerely hope that this endeavor will attract a broader range of contributors who will support me in achieving these goals._" from your https://github.com/ransome1/sleek/discussions/501

> Also let me close this thread for now, since we more or less hijacked it a little bit for something not related to the initial requirement.

Definitely, thanks

> The line break issue should be resolved in the upcoming developer preview 10 and if it is not we should rather open a separate bug task.

Yes, it was already resolved in the defective dev10 that I was able to download before you deleted it. I think however, the CRLF / LF problem after Sleek touches the file (first point from my https://github.com/ransome1/sleek/issues/68#issuecomment-1718090704 ) might be related, but I'm sure you've got it covered. Which brings me to my last point:

I suppose I will not create new issues about what I mentioned in https://github.com/ransome1/sleek/issues/68#issuecomment-1718090704 (using the defective dev10), until the new dev10 comes out? To save you some time and to not hijack this thread any more, please only reply, if you want me to create new issues right away. If I don't hear anything from you here, I would wait for the final dev10.

Thanks again @ransome1 !

**Comment by ransome1** [comment]
> I suppose I will not create new issues about what I mentioned in [#68 (comment)](https://github.com/ransome1/sleek/issues/68#issuecomment-1718090704) (using the defective dev10), until the new dev10 comes out? To save you some time and to not hijack this thread any more, please only reply, if you want me to create new issues right away. If I don't hear anything from you here, I would wait for the final dev10.

I hadn't seen this particular issue on my side, but I would propose once the Dev 10 uploaded, you give it another try on your system. And if anything weird happens, I would appreciate a separate bug report.

**Comment by ransome1** [comment]
@dnngll I tested the latest implementation on a Windows machine and couldn't reproduce the error any more. So please feel free to test it out yourself and if it still throws the error or shows some other unwanted behavior, let's open a dedicated bug report for it.

https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev10

**Comment by dnngll** [comment]
Hey @ransome1 

last week I did test all 4 files from above plus my 3 main todo-files, using `sleek-2.0.0-dev13-win.zip` `[87e66bcb3f29fdd7c7bd776d30221fa6]`. Every file worked flawlessly, and I did a lot of testing (check todo, new todo, delete todo, archive ticked todos, change todo-file from somewhere else) with all 7 files. Thank you so much!

I am on dev-13 still, but will test dev-14 (`sleek-2.0.0-dev14-win.zip` `[146e01e975161180a4064256c459c672]`, which somehow can't be found on https://github.com/ransome1/sleek/releases) from now on.

**Comment by ransome1** [comment]
@dnngll the `win.zip` should be there, here is the direct link: https://github.com/ransome1/sleek/releases/download/v2.0.0-dev14/sleek-2.0.0-dev14-win.zip

**Comment by dnngll** [comment]
Yeah I know, thanks. I've been using it since last week. I got the link through the Email GitHub sends automatically when subscribed.

I just wanted to point out that dev-14 (still) can't be found on the official release-page https://github.com/ransome1/sleek/releases for some reason.

Very much looking forward for the final 🙌

**Comment by dnngll** [comment]
Yeah I know, thanks. I've been using it since last week. I got the link through the email GitHub sends automatically when subscribed.

I just wanted to point out that dev-14 (still) can't be found on the official release-page https://github.com/ransome1/sleek/releases for some reason.

Very much looking forward for the final 🙌
